### ISSUE TITLE
MayBeConst false positives

### ISSUE THREAD

**Issue by MyDogTom** [arturbosch/detekt#808]
Version: 1.0.0.RC6-4
**Case 1**
```kotlin
    companion object {
        private val A = "asdf=${AnotherClass.staticVariable}"
    }
```
where `AnotherClass`  is a Java class with `public static String staticVariable = "asdf`

**Case 2**

```kotlin
class MyClass {
    companion object {
        val const = "${MyClass::class.java.name}.EXTRA_DETAILS"
    }
}
```

in both cases Detekt suggests to add `const`.

**Comment by schalkms** [comment]
The compiler replaces the following object `TEST` val with `"TEST test"` at compile time. `stringTemplate.hasInterpolation()` is therefore `false`.
```kotlin
var test = "test"
val code = """
  object Test {
    val TEST = "Test $test"
  }
"""
assertThat(subject.lint(code)).isEmpty()
```

@arturbosch do you have any idea how to catch this issue?

**Comment by arturbosch** [comment]
Debugged this and it seems really we can't get the necessary info oO.
I'm wondering because intellij reports the same thing we are trying to archieve.
Need to look at their inspection next.

**Comment by arturbosch** [comment]
A workaround would be to use a scanner or just simple check for the literal content not containing $ without escaped etc ...

**Comment by schalkms** [comment]
Mhhm
With PsiUtils this should work yes. Are you working on a fix @arturbosch
